clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -106,4 +114,17 @@ private boolean process(final OffsetFetchApplicationEvent event) {
         manager.addOffsetFetchRequest(event.partitions);
         return true;
     }
+
+    private boolean process(final NewTopicsMetadataUpdateRequestEvent event) {
+        metadata.requestUpdateForNewTopics();
+        return true;
+    }
+
+    private boolean process(final AssignmentChangeApplicationEvent event) {
+        Optional<RequestManager> commitRequestManger = registry.get(RequestManager.Type.COMMIT);
+        CommitRequestManager manager = (CommitRequestManager) commitRequestManger.get();
```

Comment:
If `group.id` is not specified, will `commitRequestManger` be defined? If not, we can't blindly call `get`. If yes, then we need the logic to skip offset commit when `group.id` is not specified.

Reply:
I think an isPresent check is missing here. Updated.

Reply:
FWIW: I think currently, group.id can be null as well as an empty string.  Only when the group id is null, the coordinator object won't be created.  We are planning to drop the support of empty string soon.